controller/volume_controller.go

```diff
@@ -808,12 +895,32 @@ func (vc *VolumeController) cleanupExtraHealthyReplicas(v *longhorn.Volume, e *l
 		for _, rName := range rNames {
 			r := rs[rName]
 			if r.Spec.NodeID != e.Spec.NodeID {
-				return vc.deleteReplica(r, rs)
+				if err := vc.deleteReplica(r, rs); err != nil {
+					return false, err
+				}
+				return true, nil
 			}
 		}
 	}
+	return false, nil
+}
 
-	return nil
+func (vc *VolumeController) getAutoBalancedReplicasSetting(v *longhorn.Volume) (types.ReplicaAutoBalance, error) {
+	volumeSetting := v.Spec.ReplicaAutoBalance
+	if volumeSetting != types.ReplicaAutoBalanceIgnored {
+		return volumeSetting, nil
+	}
+
+	globalSetting, err := vc.ds.GetSettingValueExisted(types.SettingNameReplicaAutoBalance)
+	if err != nil {
+		return "", err
+	}
+
+	if globalSetting == string(types.ReplicaAutoBalanceIgnored) {
+		globalSetting = string(types.ReplicaAutoBalanceDisabled)
+	}
+
+	return types.ReplicaAutoBalance(globalSetting), nil
```

Comment:
What's the difference between Ignored/Disabled?
I am guessing volumes that don't have one specified will fallback to the global setting?

Reply:
The `ignored` option is only for volume spec, when not specified in the YAML it will default to set to `ignored`, and the operation falls back to the global setting. The `ignored` option should also show as the default option in the UI. (I will create an issue for UI later).

Because we are also validating this `ignored` option against the global setting, so in any case global is set to `ignored`, it will fall back to `disabled.`

Reply:
Addressed, thanks!